### ISSUE TITLE
Grenades: AmmoInMagazine and AmmoReserve always -1 / 0

### ISSUE THREAD

**Issue by Linus4** [markus-wa/demoinfocs-golang#145]
**Describe the bug**

The `AmmoInMagazine` and `AmmoReserve` fields of the `Equipment` type always have the following values for *any grenade*, regardless of how many instances of the grenade the player actually owns. This is a problem because there is no way to tell how many flashbangs a player owns.

```
AmmoInMagazine: -1
AmmoReserve: 0
```

**To Reproduce**

Any demo I've tested so far.
https://www.hltv.org/download/demo/51879

Code:

```go
sort.Slice(weapons, func(i, j int) bool { return weapons[i].Weapon < weapons[j].Weapon })
for _, w := range weapons {
	if w.Class() == common.EqClassGrenade {
		switch w.Weapon {
		case common.EqFlash:
			// there seems to be only one flashbang in player.Weapons() even if he has two
			fmt.Println("AmmoInMagazine:", w.AmmoInMagazine)
			fmt.Println("AmmoReserve:", w.AmmoReserve)
	}
}
```

**Expected behavior**

`AmmoInMagazine` should be 1 and `AmmoReserve` should be the number of grenades the player owns of this type in addition to that.

**Library version**

v1.3.1

**Additional context**

/

**Comment by markus-wa** [comment]
It seems like grenade ammo is stored in `Player.AmmoLeft[eq.AmmoType]`.

For now you should be able to do this:
```go
for _, w := range weapons {
	if w.Class() == common.EqClassGrenade {
		switch w.Weapon {
		case common.EqFlash:
			fmt.Println("Ammo:", w.Owner.AmmoLeft[w.AmmoType])
	}
}
```

TODO:
- [ ] fix docs on `Player.AmmoLeft`
- [ ] add getter to `Weapon` for `Player.AmmoLeft`

**Comment by Linus4** [comment]
I can confirm that this works.

**Comment by BestAwperEver** [comment]
> there is no way to tell how many flashbangs a player owns

Not sure whether I should create another issue, but this also affects #130: if a player picks up a flashbang when they already have one, `WeaponTraceable()` returns already owned one instead of new.

**Comment by markus-wa** [comment]
Hmm, I think it's fine if it returns the existing grenade instance, since there is only one instance anyway.

What do you think @BestAwperEver?

**Comment by BestAwperEver** [comment]
Looks like I've missed something there :) I initially assumed that flashbang "ammo" is just a way of representing amount of owned flashbangs while both new and already owned flashbangs might have different internal IDs -- which they don't because, as I understand now, they _are_ the same weapon with actual ammo.

Well, I have faced this little issue when I was trying to distinct pickups and purchases. I mapped weapon internal IDs to their owners and on every `ItemPickup` I checked if there is already such entry in my map that corresponds to `WeaponTraceble()`'s internal ID. That worked for every weapon except for newly bought second flashbangs, because `WeaponTraceble()` always returned ID of already owned instance.

**Comment by BestAwperEver** [comment]
I will create another issue for my question since conversation seems to be a bit off-topic.

**Comment by xXxhaxmeister42xXx** [comment]
It might be worth noting that the grenade owner can be null, so when looping through Player.RawWeapons one could use Player.AmmoLeft over Owner.AmmoLeft.

```go

for _, weapon := range player.RawWeapons {
	var ammoreserve int
	if weapon.Class()==common.EqClassGrenade {
		ammoreserve = player.AmmoLeft[weapon.AmmoType];
		//instead of weapon.Owner.AmmoLeft[weapon.AmmoType]
	} else {
		ammoreserve = weapon.AmmoReserve;
	}
}
```

**Comment by markus-wa** [comment]
@xXxhaxmeister42xXx do you have a repro for that?

If so can you file it in a new issue?

Owner should never be nil when looping through `Player.Weapons()` (or `RawWeapons`)

**Comment by markus-wa** [comment]
Should be fixed in `master`

Feel free to reopen if you still see a problem.

**Comment by xXxhaxmeister42xXx** [comment]
> grenade owner can be null

> do you have a repro for that?

Well, at least i thought i had a nil pointer exception that was fixed after i checked for nil owner.
By now i had the library updated and that code block rewritten ten times and somehow fail to have my program crashing ¯\_(ツ)_/¯
Could possibly have been something else.

**Comment by markus-wa** [comment]
@xXxhaxmeister42xXx Nope, you were right! 😄

I sneakily fixed it in 6b817dbfe6f130dc38cfe355bf949a40f1c19828 (and by sneakily I mean I found it as well and completely forgot to tell you).

**Comment by xXxhaxmeister42xXx** [comment]
Ah, i see. Sneaky sneaky :D

**Comment by markus-wa** [comment]
Should be fixed with [v1.3.2](https://github.com/markus-wa/demoinfocs-golang/releases/tag/v1.3.2) 🙂
